code/__HELPERS/mobs.dm

```diff
@@ -319,9 +319,27 @@ Proc for attack log creation, because really why not
 	if(progbar)
 		qdel(progbar)
 
-/proc/isabductor(A)
+/proc/is_species(A, species_datum)
 	. = FALSE
 	if(ishuman(A))
 		var/mob/living/carbon/human/H = A
-		if(H.dna && istype(H.dna.species, /datum/species/abductor))
+		if(H.dna && istype(H.dna.species, species_datum))
 			. = TRUE
+
+/proc/isabductor(A)

```

Comment:
@coiax I don't think so, but for the time being non of the procs here are using any of that advanced functionality, so you get the speed of a single proc call with the flexibility of later converting it from a macro to a proc seamlessly